bounded-collections/src/bounded_vec.rs

```diff
@@ -289,10 +289,11 @@ impl<'a, T, S: Get<u32>> BoundedSlice<'a, T, S> {
 
 impl<T: Decode, S: Get<u32>> Decode for BoundedVec<T, S> {
 	fn decode<I: codec::Input>(input: &mut I) -> Result<Self, codec::Error> {
-		let inner = Vec::<T>::decode(input)?;
-		if inner.len() > S::get() as usize {
+		let len: u32 = <Compact<u32>>::decode(input)?.into();
+		if len > S::get() {
```

Comment:
`BoundedBTreeMap` and `BoundedBTreeSet` also suffer from this.  
Could you please also fix it there?